File: app/src/main/java/com/unis/gameplatfrom/constant/GameConstant.java
package com.unis.gameplatfrom.constant;

import com.unis.gameplatfrom.api.result.BaseCustomListResult;
import com.unis.gameplatfrom.entity.GamesEntity;
import com.unis.gameplatfrom.entity.LoginEntity;
import com.unis.gameplatfrom.model.BaseModel;
import com.unis.gameplatfrom.ui.view.BaseView;

import io.reactivex.Observable;

public interface GameConstant {

    interface Model extends BaseModel {
        Observable<BaseCustomListResult<GamesEntity>> getGameList(String uuid);
    }

    interface View extends BaseView {
        @Override
        void showLoading();

        @Override
        void hideLoading();

        @Override
        void onError(Throwable throwable);

        void onSuccess(BaseCustomListResult<GamesEntity> result);
    }

    interface Presenter {
        /**
         * 登陆
         *
         * @param uuid
         */
        void getGameList(String uuid);
    }




}

File: kotlin/src/main/java/com/unis/kotlin/constant/GameConstant.kt
package com.unis.kotlin.constant


import com.unis.kotlin.api.result.BaseCustomListResult
import com.unis.kotlin.entity.GamesEntity
import com.unis.kotlin.model.BaseModel
import com.unis.kotlin.ui.view.BaseView
import io.reactivex.Observable

interface GameConstant {

    interface Model : BaseModel{

        fun getGameList(uuid: String): Observable<BaseCustomListResult<GamesEntity>>

    }

    interface View : BaseView{

        override fun hideLoading() {

        }

        override fun showLoading() {

        }

        override fun onError(throwable: Throwable) {

        }

        fun onSuccess(result: BaseCustomListResult<GamesEntity>)


    }

    interface Persenter{

        /**
         * 获取游戏列表
         */
        fun getGameList(uuid: String)


    }





}